R/vars_manager.R

```diff
@@ -95,11 +131,36 @@ get_matrix_var <- function(obj, name, abbrev_len=DEFAULT_ABBREV_LEN) {
     ))
 }
 
-get_dataframe_var <- function(obj, name, abbrev_len=DEFAULT_ABBREV_LEN) {
+get_dataframe_var <- function(obj, name, abbrev_len=DEFAULT_ABBREV_LEN, sort_by=NULL, ascending=NULL) {
+    # sort_by should be a vector of column names, or a single string: "index"
+    # ascending should be a vector of booleans, or a single boolean
     dims <- dim(obj)
     summary <- sprintf("Size: %dx%d Memory: %s", dims[[1]], dims[[2]], human_bytes(utils::object.size(obj)))
-    abbrev <- !is.null(abbrev_len) && ncol(obj)*nrow(obj) > abbrev_len
-    obj_pre <- `if`(abbrev, obj[1:ceiling(abbrev_len/dims[[2]]),], obj)
+    abbrev <- !is.null(abbrev_len) && nrow(obj) > abbrev_len
+
+    if (is.character(sort_by) && tolower(sort_by) == "index") {
+        descending <- `if`(is.logical(ascending), !ascending, FALSE)
+        obj_sorted <- obj[order(as.numeric(rownames(obj)), decreasing=descending),]
```

Comment:
what happens when the row names are strings? `rownames(df) <- c("Row1","Row4","Row3")`